clang/runtime/dpct-rt/include/dpct/dpl_extras/dpcpp_extensions.h

```diff
@@ -566,6 +566,49 @@ class radix_sort {
   uint8_t *_local_memory;
 };
 
+/// Load linear segment items into block format across threads
+/// Helper for Block Load
+template <size_t GROUP_WORK_ITEMS, typename InputT, int ITEMS_PER_THREAD,
+          typename InputIteratorT>
+__dpct_inline__ void load_blocked(size_t linear_tid, InputIteratorT block_itr,
+                                  InputT (&items)[ITEMS_PER_THREAD]) {
+#pragma unroll
+  for (int ITEM = 0; ITEM < ITEMS_PER_THREAD; ITEM++) {
+    if ((linear_tid * ITEMS_PER_THREAD) + ITEM < GROUP_WORK_ITEMS) {
+      items[ITEM] = block_itr[(linear_tid * ITEMS_PER_THREAD) + ITEM];
+    }
+  }
+}
+
+template <size_t GROUP_WORK_ITEMS, typename InputT, int ITEMS_PER_THREAD,
+          typename InputIteratorT>
+__dpct_inline__ void load_striped(size_t linear_tid, InputIteratorT block_itr,
+                                  InputT (&items)[ITEMS_PER_THREAD]) {
+#pragma unroll
+  for (int ITEM = 0; ITEM < ITEMS_PER_THREAD; ITEM++) {
+    if (linear_tid + (ITEM * ITEMS_PER_THREAD) < GROUP_WORK_ITEMS) {
+      items[ITEM] = block_itr[linear_tid + (ITEM * ITEMS_PER_THREAD)];
+    }
+  }
+}
+
+template <size_t GROUP_WORK_ITEMS, typename Item, typename InputT,
+          int ITEMS_PER_THREAD, typename InputIteratorT>
+__dpct_inline__ void load_subgroup_striped(const Item &item, size_t linear_tid,
+                                           InputIteratorT block_itr,
+                                           InputT (&items)[ITEMS_PER_THREAD]) {
+
+  // int tid = linear_tid & 1;
+  // int wid = linear_tid >> 5;
+  // int warp_offset = wid * ITEMS_PER_THREAD;
+#pragma unroll
+  for (int ITEM = 0; ITEM < ITEMS_PER_THREAD; ITEM++) {
+    new (&items[ITEM])
+        InputT(block_itr[item.get_sub_group().get_local_range()[0] +
+                         linear_tid + (ITEM * ITEMS_PER_THREAD)]);
```

Comment:
A similar comment to above with regards to the stride. In this case, I believe instead of `ITEMS_PER_THREAD` it should be the subgroup size which